### ISSUE TITLE
File Upload requires a file on disk

### ISSUE THREAD

**Issue by Oralordos** [stripe/stripe-go#374]
Currently the `FileUploadParams` requires that the file you are going to upload is of type `*os.File`. Looking through the code, all that you actually want is the filename, and an `io.Reader`.

If the file is being uploaded by a user to your server, for a managed account for example, there is no file on disk. If you are on AppEngine especially, where you have no write access to the disk, you cannot send the user's identity document using this library.

To keep old code working, a transition period could be put in where the file is supported, as well as the reader and filename codepath, similar to how the `Account` and `StripeAccount` fields are being used now.

In short, `FileUploadParams` should be updated to take a filename and a reader instead of a file.
